### DISCUSSION TITLE
OrmLite with in-memory Sqlite on AWS Linux machine

### DISCUSSION THREAD

**Discussion by gerdaditeriha** [ServiceStack/Discuss#46] [Q&A]
Hi. I am using OrmLite to run some tests using Sqlite in-memory option. All is good, when running locally on my windows machine. But the tests fail on AWS Linux machine(using **amazonlinux2-aarch64-standard:2.0** image)

This is how I create connection

`dbFactory = new OrmLiteConnectionFactory(":memory:", new SqliteOrmLiteDialectProvider());`
...
`dbFactory .OpenDbConnection()`

On my AWS Linux machine I get this error:
System.DllNotFoundException : Unable to load shared library 'SQLite.Interop.dll' or one of its dependencies.
![image](https://user-images.githubusercontent.com/83020163/209109954-6da2870c-6d43-4777-8345-d405a3fde618.png)

I have tried it also on Ubunut VM, there is works, just like on my local windows machine. Anyone has issues running on this particular AWS image?

These are the OrmLite libraries I have
![image](https://user-images.githubusercontent.com/83020163/209112689-9855a21a-d9cd-4eeb-8c04-b37781009a58.png)

Following other suggestion I have installed bunch of sqlite libraries
![image](https://user-images.githubusercontent.com/83020163/209112919-369bb461-3e45-4f56-be3a-d9435461be70.png)

None of those helped

**Comment by mythz** [comment]
You have 2 [OrmLite Sqlite packages referenced](https://docs.servicestack.net/ormlite/installation#sqlite), you should only be referencing the one implementation you're using. 

If this is for ARM server you'll have the best chance with Microsoft.Data.Sqlite provider, i.e:

```xml
<PackageReference Include="ServiceStack.OrmLite.Sqlite.Data" Version="6.*" />
```

**Reply by gerdaditeriha** [reply]
Thank you for the suggestion. Setting version to 6.* and deleting other OrmLite.Sqlite library did not help though :(

**Reply by mythz** [reply]
It should definitely help with this error since you should no longer be using or referencing the package that produces this StackTrace. Whether or not the `Microsoft.Data.Sqlite` provider supports this platform would be a different issue.
